elasticdl/python/master/master.py

```diff
@@ -430,3 +436,21 @@ def _create_instance_manager(self, args):
             )
 
         return instance_manager
+
+    def _check_timeout_tasks(self):
+        while True:
+            doing_tasks = self.task_d._doing.copy()
+            cur_time = time.time()
+            threshold = (
+                2 * self.master_servicer.get_average_task_complete_time()
+            )
+            for task_id, (worker_id, task, get_time) in doing_tasks.items():
+                if (cur_time - get_time) > threshold:
+                    self.logger.info(
+                        "worker %d timeout, relaunch it" % worker_id
+                    )
+                    self.task_d.recover_tasks(worker_id)
```

Comment:
Do we need to set the recovery times for a task? IMO, the task should fail if the times to recover it due timeout exceed a threshold. Now, we will recover a bad task repeatedly.

Reply:
Good idea. It's another topic, and will be fixed in next PR.